server/checks/PullRequestTasks.js

```diff
@@ -0,0 +1,78 @@
+import Check, { getPayloadFn } from './Check'
+import { PULL_REQUEST } from '../model/GithubEvents'
+import { logger } from '../../common/debug'
+
+const CHECK_TYPE = 'pullrequesttasks'
+const CONTEXT = 'zappr/pr/tasks'
+const info = logger(CHECK_TYPE, 'info')
+const error = logger(CHECK_TYPE, 'error')
+const debug = logger(CHECK_TYPE)
+const createStatePayload = getPayloadFn(CONTEXT)
+
+export function countOpenTasks(prBody, repoName = "", prNo = "") {
+  const regex = /(-|\*) \[(\s|x)\]/g;
+  let count = 0;
+  let match;
+
+  while ((match = regex.exec(prBody)) !== null) {
+    // This is necessary to avoid infinite loops with zero-width matches
+    if (match.index === regex.lastIndex) {
+      regex.lastIndex++;
+    }
+
+    let task = match[0];
+    let marker = match[2];
+
+    debug(`${repoName}#${prNo}: Processing task: ${task}`);
+
+    if (marker === " ") {
+      count++;
+    }
+  }
+
+  info(`${repoName}#${prNo}: PR has ${count} open tasks.`);
+
+  return count;
+}
+
+export default class PullRequestTasks extends Check {
+  static TYPE = CHECK_TYPE;
+  static CONTEXT = CONTEXT;
+  static HOOK_EVENTS = [PULL_REQUEST];
+  static NAME = 'Pull request tasks check'
+
+  constructor(github) {
+    super()
+    this.github = github;
+  }
+
+  async execute(config, hookPayload, token) {
+    const {action, repository, number, pull_request} = hookPayload
+    const repoOwner = repository.owner.login
+    const repoName = repository.name
+    const fullName = repository.full_name
+    let status;
+
+    try {
+      if (pull_request.state === 'open' && ['opened', 'edited', 'reopened'].indexOf(action) !== -1) {
+        let openTaskCount = countOpenTasks(pull_request.body, repoName, number);
```

Comment:
Ah, sorry if my suggestion was misleading. Putting the fullName inside there is a good idea as it's otherwise hard to distinguish different repos with the same name.